### PR TITLE
Handle SStream overflow by truncating appends

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [ ] I've documented or updated the documentation of every API function and struct this PR changes.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)

**Detailed description**

This enforces null terminated buffer and behaves better as it truncates properly in case there's not enough space. im fine for not merging this but i prefer a consistent behaviour than an assert with a macro. but thats just up to decide the behaviour users expects from this corner case
...

**Test plan**

there's a test attached
...

**Closing issues**

...

### CODE REVIEW

File: SStream.c [hunk 1]
```diff
@@ -231,21 +254,25 @@ void SStream_concat(SStream *ss, const char *fmt, ...)
 	SSTREAM_RETURN_IF_CLOSED(ss);
 	va_list ap;
 	int ret;
+	size_t remaining = SStream_remaining(ss);
+	if (remaining == 0) {
+		return;
+	}
```
Comment (Rot127) [inline SStream.c:260]:
```suggestion
	CS_ASSERT_RET(remaining > 0)
```

Otherwise there is no warning for this broken case.

**Review by Rot127 — CHANGES_REQUESTED** [review]
For sure way cleaner! Thanks.
One nitpick, otherwise lgtm.

File: SStream.c [hunk 2]
```diff
@@ -255,9 +255,7 @@ void SStream_concat(SStream *ss, const char *fmt, ...)
 	va_list ap;
 	int ret;
 	size_t remaining = SStream_remaining(ss);
-	if (remaining == 0) {
-		return;
-	}
+	CS_ASSERT_RET(remaining > 0)
```
Comment (Rot127) [inline SStream.c]: Missing `;`.

(Can make a suggestion. Github is broken again for whatever reason.)

**Review by Rot127 — CHANGES_REQUESTED** [review]

File: SStream.c [hunk 3]
```diff
@@ -231,21 +254,23 @@ void SStream_concat(SStream *ss, const char *fmt, ...)
 	SSTREAM_RETURN_IF_CLOSED(ss);
 	va_list ap;
 	int ret;
+	size_t remaining = SStream_remaining(ss);
+	CS_ASSERT_RET(remaining > 0)
```
Comment (trufae) [inline SStream.c]:
```suggestion
	CS_ASSERT_RET(remaining > 0);
```

File: SStream.c (hunk 2 above)
Comment (trufae) [reply SStream.c]: Done

File: SStream.c [hunk 4]
```diff
@@ -231,21 +254,23 @@ void SStream_concat(SStream *ss, const char *fmt, ...)
 	SSTREAM_RETURN_IF_CLOSED(ss);
 	va_list ap;
 	int ret;
+	size_t remaining = SStream_remaining(ss);
+	CS_ASSERT_RET(remaining > 0);
```
Comment (Rot127) [inline SStream.c]:
```suggestion
	if (remaining == 0) {
		return;
	}
```

🙄 Of course I forgot about your unit test.
Overflow test has priority, so lets reverse it.

I really need to make myself more time for reviews...

**Comment by Rot127** [comment]
@trufae Can you please run clang-format real quick. Otherwise lgtm and I would merge.

**Review by Rot127 — APPROVED** [review]

**Comment by Rot127** [comment]
Cherry-picked your changes here: https://github.com/capstone-engine/capstone/pull/2921

Thanks a lot!
